<launch>
     
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle]"/>
    <arg name="real_x" default="1.0"/>
    <arg name="real_y" default="2.0"/>
    <arg name="fake_x" default="1.2"/>
    <arg name="fake_y" default="2"/>
    <arg name="world_location" default="$(find turtlebot3_gazebo)/worlds/turtlebot3_world.world"/>
    <arg name="map_file" default="$(find multi_robot_nav)/config/map.yaml"/>

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch" />

  <!-- Map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
  </node>

  <!-- AMCL -->
  <include file="$(find multi_robot_nav)/launch/single_movebase_files/amcl.launch">
  <arg name="initial_pose_x" value="$(arg fake_x)"/>
  <arg name="initial_pose_y" value="$(arg fake_y)"/>
  </include>

  <!-- move_base -->
    <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base node -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find multi_robot_nav)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find multi_robot_nav)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find multi_robot_nav)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
  
  <!-- Commented out this node since it is already launched in the navigation launch file or in included files>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" type="double" value="50.0" />
  </node> -->


  <!-- Gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_location)"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model turtlebot3_$(arg model) -x $(arg real_x) -y $(arg real_y) -z 0.0 -param robot_description" />
  
  <node name="rviz" pkg="rviz" type="rviz" required="true"
     args="-d $(find multi_robot_nav)/rviz/myconfig.rviz"/>

</launch>
